supabase/migrations/17_ops_catalogs.sql

```diff
@@ -0,0 +1,128 @@
+This will contain the placeholder TENANT, which will be replaced by the actual tenant name prior to publication.
+The value for this is generated by running `flowctl raw bundle` on the catalog specs in the `ops-catalog/` directory.
+Typically, this will actually get updated by using ops-catalog/generate-migration.sh';
+
+create function internal.create_ops_publication(tenant_prefix text, ops_user_id uuid)
+returns flowid as $$
+declare
+	new_draft_id flowid := internal.id_generator();
+	publication_id flowid := internal.id_generator();
+	ops_template json;
+	
+begin
+	select replace(bundled_catalog::text, 'TENANT', rtrim(tenant_prefix, '/'))::json into ops_template from ops_catalog_template where id =  '00:00:00:00:00:00:00:00';
+
+	insert into drafts (id, user_id, detail) values (new_draft_id, ops_user_id, 're-publishing ops catalog');
+	
+	-- Start by adding a deletion for all currently live ops catalog items.
+	insert into draft_specs (id, draft_id, catalog_name, spec_type, spec)
+		select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			catalog_name,
+			null as spec_type,
+			null as spec
+		from live_specs
+		where catalog_name like ('ops/' || tenant_prefix || '%');
+
+	insert into draft_specs (id, draft_id, catalog_name, spec_type, spec)
+		(
+		select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'capture'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'captures'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'collection'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'collections'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'materialization'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'materializations'))
+		union all select
+			internal.id_generator() as id,
+			new_draft_id as draft_id,
+			"key" as catalog_name,
+			'test'::catalog_spec_type as spec_type,
+			value as spec
+		from json_each(json_extract_path(ops_template, 'tests'))
+		)
+		on conflict (draft_id, catalog_name) do update
+			set spec_type = excluded.spec_type, spec = excluded.spec;
+
+	insert into publications (id, user_id, draft_id) values (publication_id, ops_user_id, new_draft_id);
+	return publication_id;
+end;
+$$ language plpgsql
+security definer;
+
+comment on function internal.create_ops_publication is '
+Creates a new publication of the ops catalog template for a specific tenant.
+This publication will include all specs from the bundled_catalog in ops_catalog_template.
+Any _other_ specs under the ops/<tenant>/ prefix will be deleted by this publication.
+';
+
+
+create function internal.update_ops_catalog_template(bundled_catalog_arg json)
```

Comment:
nit: this function doesn't carry a lot of water, IMO, and its body should be lifted to the migration script -- which, as i understand things, would be the only non-test place we ever actually update this. I don't feel very strongly though if you'd rather keep it (but in that case it should probably have test coverage).